modules/lotamePanoramaIdSystem.js

```diff
@@ -148,6 +147,42 @@ export const lotamePanoramaIdSubmodule = {
    */
   name: MODULE_NAME,
 
+  /**
+   * Find the root domain
+   * @param {string} fullDomain
+   * @return {?string}
+   */
+  findCookieDomain: function (fullDomain) {
+    if (!storage.cookiesAreEnabled()) {
+      return fullDomain;
+    }
+
+    const domainParts = fullDomain.split('.');
+    if (domainParts.length == 2) {
+      return fullDomain;
+    }
+    let rootDomain;
+    let continueSearching;
+    let startIndex = -2;
+    const TEST_COOKIE_NAME = 'lotame_domain_check';
+    const TEST_COOKIE_VALUE = 'writeable';
+    do {
+      rootDomain = domainParts.slice(startIndex).join('.');
+      let expirationDate = new Date(
+        utils.timestamp() + 10 * 1000
+      ).toUTCString();
+      storage.setCookie(TEST_COOKIE_NAME, TEST_COOKIE_VALUE, expirationDate, 'Lax', `${rootDomain}`, undefined);
+      const value = storage.getCookie(TEST_COOKIE_NAME, undefined);
+      if (value === TEST_COOKIE_VALUE) {
+        continueSearching = false;
+      } else {
+        startIndex += -1;
+        continueSearching = Math.abs(startIndex) <= domainParts.length;
+      }
+    } while (continueSearching);
+    return rootDomain;
+  },
```

Comment:
there's at least one or two other id modules that are doing essentially the same thing -- finding the root domain at which to write the 1p cookie. what are people's thoughts on moving this to a method in `utils` or `userId` module that can be re-used rather than repeated in multiple modules?

cc @patmmccann @Fawke @jdwieland8282 @robertrmartinez @pycnvr

Reply:
@smenzer Yes, I like your idea.

Reply:
@idettman do you have thoughts here? is this something we can/should break out to be a common userid (or even common prebid global) method?

Reply:
Are you all thinking this should be done in context of this PR or separately?

Reply:
I think it probably makes the most sense to break this out into a separate PR, then you can use this PR to apply it to the Lotame ID system

Reply:
Alright, I will give that shot

Reply:
That part is attempted in https://github.com/prebid/Prebid.js/pull/6124

Reply:
@markaconrad thanks for putting in #6124! I assume once that goes through, you would remove this block and on line 206 below you'd use the user id function instead of `this` right?

Reply:
You are welcome! Yes, that is the plan